src/libnnstreamer-edge/nnstreamer-edge-internal.c

```diff
@@ -1260,68 +1273,79 @@ nns_edge_request (nns_edge_h edge_h, nns_edge_data_h data_h)
     return NNS_EDGE_ERROR_INVALID_PARAMETER;
   }
 
-  conn_data = _nns_edge_get_connection (eh, eh->client_id);
-  if (!conn_data || !_nns_edge_check_connection (conn_data->sink_conn)) {
-    nns_edge_loge ("Failed to request, connection failure.");
-    nns_edge_unlock (eh);
-    return NNS_EDGE_ERROR_CONNECTION_FAILURE;
-  }
-
-  _nns_edge_cmd_init (&cmd, _NNS_EDGE_CMD_TRANSFER_DATA, eh->client_id);
+  /** @todo update code for each connect type */
+  switch (eh->connect_type) {
+    case NNS_EDGE_CONNECT_TYPE_TCP:
```

Comment:
How about adding hybrid connect type since it operates same with TCP.
```suggestion
    case NNS_EDGE_CONNECT_TYPE_TCP:
    case: NNS_EDGE_CONNECT_TYPE_HYBRID:
```

Reply:
added